File: Projects/codechef/MARBLES.java
package codechef;

import java.util.Scanner;

class MARBLES {
    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int t = stdin.nextInt();
        while (t-- != 0) {
            int n = 0, k = 0;
            if (stdin.hasNextInt()) {
                n = stdin.nextInt();
            }
            if (stdin.hasNextInt()) {
                k = stdin.nextInt();
            }
            if (n == k) {
                System.out.println(1);
            } else {
                n = n - k;
                long ans = 1;
                for (int i = 1; i < k; i++) {
                    ans = ans * (n + i) / i;
                }
                System.out.println(ans);
            }
        }
    }
}

File: Projects/codechef/MARBLES.kt
package codechef

import java.util.*

object MARBLES {
    private fun choose(n: Int, k: Int): Long {
        var res: Long = 1
        for (i in 0 until k) {
            res = res * (n - i) / (i + 1)
        }
        return res
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        var T = stdin.nextInt()
        while (T-- > 0) {
            val N = stdin.nextInt()
            val K = stdin.nextInt()
            println(choose(N - 1, K - 1))
        }
        stdin.close()
    }
}